### PR TITLE
Add table on how to link op values to data schema related terms

### LINKED ISSUE #1469: Mapping of Data Schemas to operations
The comment at https://github.com/w3c/wot-thing-description/issues/1324#issuecomment-1005841116 made me realize that we do not explain very well how data schemas of an affordance maps to the operations. We implicitly know this, i.e. `subscription` in events maps to the `subscribeevent` operation. However, I think that the spec needs a nice table that says that for this operation, the data schema (or the payload) is defined by:
- these terms in the affordance
- protocol related information in the forms
- lack of both means empty payload

### PR DESCRIPTION
**Pull request by egekorkan**
fixes #1469 

I have also moved a paragraph above and added additional subsections to improve readabilty

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1493.html" title="Last updated on May 18, 2022, 3:40 PM UTC (9e2b835)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1493/f6ab9c0...9e2b835.html" title="Last updated on May 18, 2022, 3:40 PM UTC (9e2b835)">Diff</a>

### CODE REVIEW

**Comment by egekorkan** [comment]
It might be worthwile to consider moving this table somewhere else but it fits well the list of op values table

File: index.html [hunk 1]
```diff
@@ -1209,16 +1209,16 @@ <h2>Core Vocabulary Definitions</h2>
                 names are used in a <code>security</code>
                 name-value pair.</td><td>mandatory</td><td><a>Map</a> of <a href="#securityscheme"><code>SecurityScheme</code></a></td></tr>
 <tr class="rfc2119-table-assertion" id="td-vocab-profile--Thing"><td><code>profile</code></td><td>Indicates the WoT Profile mechanisms followed by this Thing Description and the corresponding Thing implementation.</td><td>optional</td><td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#anyURI"><code>anyURI</code></a> or <a>Array</a> of <a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#anyURI"><code>anyURI</code></a></td></tr>
+<tr class="rfc2119-table-assertion" id="td-vocab-schemaDefinitions--Thing"><td><code>schemaDefinitions</code></td><td>Set of named data schemas.
```
Comment (mlagally) [inline index.html:1212]: This is a normative assertion which is unrelated to the scope of the PR.

File: index.html [hunk 2]
```diff
@@ -1209,16 +1209,16 @@ <h2>Core Vocabulary Definitions</h2>
...
+                To be used in a <code>schema</code>
+                name-value pair inside an 
+                <code>AdditionalExpectedResponse</code> object.</td><td>optional</td><td><a>Map</a> of <a href="#dataschema"><code>DataSchema</code></a></td></tr>
 <tr class="rfc2119-table-assertion" id="td-vocab-uriVariables--Thing"><td><code>uriVariables</code></td><td>Define URI template variables according to [[RFC6570]]  as collection
                 based on DataSchema declarations. The Thing level <code>uriVariables</code> can be used in Thing-level 
                 <code>forms</code> or in Interaction Affordances.
                 The individual variables DataSchema cannot be an ObjectSchema or an ArraySchema.
                 If the same variable is both declared in Thing-level <code>uriVariables</code> and in Interaction Affordance level,
-                 the Interaction Affordance level variable takes precedence.</td><td>optional</td><td><a>Map</a> of <a href="#dataschema"><code>DataSchema</code></a></td></tr>
-<tr class="rfc2119-table-assertion" id="td-vocab-schemaDefinitions--Thing"><td><code>schemaDefinitions</code></td><td>Set of named data schemas.
```
Comment (mlagally) [inline index.html:1218]: These changes remove normative language and are unrelated to the scope of this PR.

File: index.html [hunk 3]
```diff
@@ -2349,6 +2349,94 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
           </table>
         </div>
 
+        <p>
+          A <a>Thing Description</a> of a WoT producer may have multiple forms entries with, e.g., different 
+          protocol and/or content types declarations that a <a>Consumer</a> could possibly support. In that case 
+          the Consumer may choose any form entry that works (e.g., the protocol and content type is supported) for them.
```
Comment (mlagally) [inline index.html:2355]: should that be an assertion?

File: index.html [hunk 4]
```diff
@@ -2349,6 +2349,94 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
           </table>
         </div>
 
+        <p>
+          A <a>Thing Description</a> of a WoT producer may have multiple forms entries with, e.g., different 
+          protocol and/or content types declarations that a <a>Consumer</a> could possibly support. In that case 
+          the Consumer may choose any form entry that works (e.g., the protocol and content type is supported) for them. 
+          When one form is chosen, it is expected that the <a>Consumer</a> will continue to use it as long as possible 
+          for every new interaction with the WoT producer.
```
Comment (mlagally) [inline index.html:2357]: What do we mean by that? Under which circumstances is it no longer posible to use a form? What should happen in this case?

File: index.html [hunk 5]
```diff
@@ -2349,6 +2349,94 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
           </table>
         </div>
 
+        <p>
+          A <a>Thing Description</a> of a WoT producer may have multiple forms entries with, e.g., different 
+          protocol and/or content types declarations that a <a>Consumer</a> could possibly support. In that case
```
Comment (mlagally) [inline index.html:2354]: What should happen, when there are multiple entries for the same protocol and operation?
Is there a precedence?

**Review by mlagally — COMMENTED** [review]
The PR contains unrelated changes to normative language.

File: index.html (hunk 1 above)
Comment (egekorkan) [reply index.html:1212]: Happens due to auto generation script, can be ignored

**Comment by egekorkan** [comment]
> The PR contains unrelated changes to normative language.

I saw this comment in many places. This keeps happening and it's not under my control

File: index.html (hunk 3 above)
Comment (egekorkan) [reply index.html:2355]: Good point. I think that the second one is indeed an assertion

File: index.html (hunk 4 above)
Comment (egekorkan) [reply index.html:2357]: I think this was a discussion somewhere else as well. In general, we do not specify consumer behavior in the TD spec.

Comment (egekorkan) [reply index.html:2357]: So it might be a good point to remove this or have a separate section and reference to that

**Comment by egekorkan** [comment]
@mlagally I have just realized that some comments are not about what I wrote. If you go to https://w3c.github.io/wot-thing-description/#expectedresponse and scroll **UP** one paragraph, you will see this

**Pull request merged**